### DISCUSSION TITLE
MHP30 - struggling to sustain a temperature below 100°C

### DISCUSSION THREAD

**Discussion by akhilakasturi1** [Ralim/IronOS#1979] [Q&A]
Hello! I just got a new MHP30 that I wanted for a use-case to maintain a constant temperature below 100°C (after checking #1848). My new MHP30 arrived and after browsing for a bit, I flashed a hex file from the latest build following #1812. But now when the temperature is set to 50°C, the temperature on my MHP30 rise to 62°C and then slowly starts to drop down to room temperature. Is this expected behavior? I want to be able to use the MHP30 where I can set a temperature (say 37°C) and have it hold constant for some time. Any help is appreciated. Thank you!

**Comment by Ralim** [comment]
Hia,
Yes this absolutely should work.
I'm going to guess this is from the newer tips having a very different thermal mass than the old ones.
Ideally #1961 should help here, you could try a build from that PR potentially.

What is your use case?

Another way to dampen out the response is to use a lower wattage usb-c power supply (not optimal, but could help in a pinch at least for now)

**Reply by akhilakasturi1** [reply]
Hi Ralim, 

I am trying to use MHP30 as a hotplate to calibrate my piezoresistive pressure sensor at different temperatures (all below 100°C). I will also have a temperature sensor on that board and I want to perform temperature compensation on my pressure sensor from this data. I found the MHP30 desirable for my use-case since I can power it using USB-PD, so I can place it in a vacuum chamber without worrying about power cord feedthrough. 

FYI, I am using the [UGREEN powerbank](https://www.amazon.com/UGREEN-Power-20000mAh-Portable-Charger/dp/B0C3GTMX5M/ref=pd_ci_mcx_pspc_dp_d_2_t_1?pd_rd_w=Z2O5u&content-id=amzn1.sym.568f3b6b-5aad-4bfd-98ee-d827f03151e4&pf_rd_p=568f3b6b-5aad-4bfd-98ee-d827f03151e4&pf_rd_r=SM3439PVNXZKDZDMHB6K&pd_rd_wg=bfufJ&pd_rd_r=6de3e280-6aae-4278-8292-39837e71ea11&pd_rd_i=B0C3GTMX5M) to power the MHP30. 

I am a beginner for github, so forgive me if this is obvious, but I looked through the PR for #1961 and I see that all the artifacts from that build are expired. Do you know where I can download the relevant hex file for MHP30  from instead?

**Reply by Ralim** [reply]
Ah, expired assets are a thing yeah; I'll manually run a build for you in a bit and upload it in a bit when I get some time.

For that use case you can possibly also limit the maximum wattage in advanced settings to make the unit a bit "de-tuned" so it ramps slower if need be.

**Reply by akhilakasturi1** [reply]
Hi Ralim, 
Yes, I will trying experimenting with limiting power and report here. Also, thank you very much for helping me with the build!
